python/cudnn/gemm/cutedsl/grouped/weight_only_nvfp4/api.py

```
@@ -0,0 +1,320 @@
+class GroupedGemmWeightOnlyNvfp4(APIBase):
+    """Precompiled standalone grouped W4A16 projection.
+
+    The checkpoint-native inputs are packed E2M1 weights in ``uint8`` storage,
+    E4M3 group-16 weight scales, BF16 routed tokens, starts-only INT32 expert
+    offsets, and one FP32 output factor per expert. The kernel writes a
+    preallocated BF16 output and owns no workspace.
+    """
+
+    def __init__(
+        self,
+        sample_routed_tokens: torch.Tensor,
```

Ruff (0.16.3)

[error] Undefined name `torch`

(F821)